### PR TITLE
add override logic using ThemeProvider to createSvgIcon factory

### PR DESCRIPTION
**Pull request by tomi-msft**
#### Pull request checklist

- [x] Addresses an existing issue: Part of #15758
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR adds logic to the `createSvgIcon` svg icon factory in order to leverage the ThemeProvider to provide the new svg icon packages with overriding capability, as outline in #15434 

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/theme/src/types/Theme.ts [hunk 1]
```diff
@@ -133,6 +134,8 @@ export interface Theme extends IScheme {
    */
   components?: ComponentsStyles;
 
+  icons?: IIconSubset;
```
Comment (xugao) [inline packages/theme/src/types/Theme.ts:137]: I don't think this belongs to the theme. I would prefer to have a separate "provider" which purely serving the purpose of overriding icons - `IconProvider` for providing &` useIcons()` for consuming.

Comment (ecraig12345) [reply packages/theme/src/types/Theme.ts:137]: A separate IconProvider would also solve the circular dependency issue probably.

Comment (ecraig12345) [reply packages/theme/src/types/Theme.ts:137]: Another option for addressing the circular dep would be moving IIconSubset out of style-utilities and into some other package (but not sure which one)

Comment (ecraig12345) [reply packages/theme/src/types/Theme.ts:137]: Actually I take both of those comments back somewhat--going to make another comment on the test file.

**Comment by tomi-msft** [comment]
Closing this PR and making a new one with an `<IconProvider />` object: [New PR](https://github.com/microsoft/fluentui/pull/16519)

**Pull request closed without merging**